### ISSUE TITLE
Support for `{ include = "*", from = "path" }` syntax

### ISSUE THREAD

**Issue by kevinvalk** [DavidVujic/poetry-multiproject-plugin#40]
**Is your feature request related to a problem? Please describe.**
We are using packages using the syntax `{ include = "*", from = "path" }` which is not supported. Currently `build-project` will throw a `No such file or directory:` error.

**Describe the solution you'd like**
It should behave as poetry behaves. For a * include it "simply" produces a pth file to add that path. So I think it would be sufficient to add ALL folders specified in the `from` directory (maybe filter based on presence of pyproject.toml?) to the temp directory.

**Describe alternatives you've considered**
Rewriting our packages to not use * in include, but given our specific setup this is almost impossible.

**Additional context**
-

**Comment by DavidVujic** [comment]
Thank you for reporting this @kevinvalk!

Do you have an example of your folder structure? This would help me to investigate and find a solution.

**Comment by DavidVujic** [comment]
Alternatively, would this be an option @kevinvalk ?

`{ include="the_folder", from = "../../" }`

**Comment by DavidVujic** [comment]
Closing this one, please re-open if this is still an issue.

**Comment by kevinvalk** [comment]
Hei, would like to re-open this one as usage of wildcard is a real use case and cannot be rewritten into include from directory.

The example from the official docs will fail when using build-project (https://python-poetry.org/docs/pyproject/#packages). This throws `[Errno 2] No such file or directory: '/Users/a/b/c/d/scripts/*.py'`

```toml
[tool.poetry]
# ...
packages = [
    { include = "my_package" },
    { include = "extra_package/**/*.py" },
]
```

I guess to fix we need to scan for files first using [glob.glob()](https://docs.python.org/3.11/library/glob.html#glob.glob). Then use this to copy all matches "things" either using normal file copy or `copytree` for matching directories. The only thing that might be "weird" is to where do you copy? 

https://github.com/DavidVujic/poetry-multiproject-plugin/blob/8d650f94b51f3a6d3ef79af51aba5ba0db81b192/poetry_multiproject_plugin/components/project/packages.py#L9-L33

When using top level wildcard (for example `{ include = "*", from = "src" }`), poetry maps them using .pth files (https://docs.python.org/3/library/site.html). In other words, the "from" path is added to `sys.path` verbatim. So for the multiproject-plugin it really is just `cp -r <from>/* <destination>`. Finally, for more complex glob patterns I am not 100% sure how it should behave. My gut feeling is that you take the `glob.glob(to)` which are file paths that are paths relative to `from` and copy those to `destination/<path>`. So for example to `{ include = "scripts/*.py" }` that matches `<from>/scripts/some-script.py` should be copied to `<destination>/scripts/some-script.py`.

P.S. After rereading your README.md can it be that within the multiproject-plugin the semantics of the "include" field in "packages" is not exactly the same as from poetry? Especially because the `copy_packages` does not even check for the existence of the file/directory at `<from>/<include>`. Poetry has the optional `to` parameter for more control. Maybe most of the conceptual problems are solved when also adding support of the `to` parameter.

**Comment by DavidVujic** [comment]
Hi @kevinvalk, thank you for explaining this scenario.

One thing I don't understand is why not use the standard `build` command for this? Do you have any examples where the MultiProject command would be used, so I can understand the use case better?

**Comment by DavidVujic** [comment]
Also, thank you for letting me know about the `to` option in Poetry. I didn't know that was added in the latest version of Poetry. Good news!

**Comment by DavidVujic** [comment]
Ok! So the microservices use one or more packages, along with their own code maybe. The structure of the package includes then would both have the `from = `../../` and some includes with `*` that are pointing to a folder within the service itself. If I understand it right, that is why you need the MultiProject `build-project` command?

**Comment by DavidVujic** [comment]
I have published a new version of this plugin: v1.5.0

This version will allow local project-specific code to have wildcards in it.

Example:
```
packages = [{ include = "extra_package/**/*.py" }],
```

If you include code using relative paths traversing upwards and outside of the project boundary, it is still expected to be pointing at a directory and not with wildcards. I hope this will help!
